lib/puppet/util/windows/service.rb

```diff
@@ -360,13 +360,16 @@ def resume(service_name)
     # @param [:string] service_name name of the service to query
     # @return [string] the status of the service
     def service_state(service_name)
-      status = nil
+      state = nil
+      current_state = nil
       open_service(service_name, SC_MANAGER_CONNECT, SERVICE_QUERY_STATUS) do |service|
-        status = query_status(service)
+        query_status(service) do |status|
+          current_state = status[:dwCurrentState]
+          state = SERVICE_STATES[current_state]
+        end
       end
-      state = SERVICE_STATES[status[:dwCurrentState]]
       if state.nil?
-        raise Puppet::Error.new(_("Unknown Service state '%{current_state}' for '%{service_name}'") % { current_state: status[:dwCurrentState].to_s, service_name: service_name})
+        raise Puppet::Error.new(_("Unknown Service state '%{current_state}' for '%{service_name}'") % { current_state: current_state.to_s, service_name: service_name})
```

Comment:
This should probably be `state` to avoid printing out the integer. I've run into that before.